### PR TITLE
Move special-case and tags from hard code to ui setting

### PR DESCRIPTION
**Pull request by benfoley**
This makes ui settings for the `special-cases` and `translation_tags` values that are used in cleaning the annotations.

It required added a ui_format property to the import settings config stuff so that more control could be had over using types of input components. The type of input the ui used was determined by the type of ui setting set, but this was limited to str, int and list. I wanted the UI for `special-cases` and `translation_tags` to be text area, and thought it best to have some mechanism for specifying that when defining the ui settings in elan transformer. 

Corresponding gui work is https://github.com/CoEDL/elpis-gui/pull/85

### CODE REVIEW

**Comment by benfoley** [comment]
This moves

File: elpis/transformer/__init__.py [hunk 1]
```diff
@@ -448,13 +448,14 @@ def import_setting(self, key, type, default=None, display_name=None, description
...
             'type': self._type_to_str(type),
+            'ui_format': ui_format,
             'display_name': display_name,
             'description': description
         }
         self._import_ui_type_config[key] = 'setting'
         self._import_ui_order_config.append(key)
     
-    def export_setting(self, key, type, default=None, display_name=None, description=None):
+    def export_setting(self, key, type, ui_format=None, default=None, display_name=None, description=None):
```
Comment (nicklambourne) [inline elpis/transformer/__init__.py:458]: Can we get types on these parameters?

Comment (nicklambourne) [reply elpis/transformer/__init__.py:458]: And make the comment params match the actual ones

File: elpis/transformer/__init__.py [hunk 2]
```diff
@@ -498,7 +500,7 @@ def general_setting_title(self, title):
         self.export_setting_title(title)
         return
 
-    def general_setting(self, key, type, default=None, display_name=None, description=None):
+    def general_setting(self, key, type, ui_format=None, default=None, display_name=None, description=None):
```
Comment (nicklambourne) [inline elpis/transformer/__init__.py:503]: types and comment

File: elpis/engines/common/objects/interface.py [hunk 3]
```diff
@@ -234,28 +232,23 @@ def list_models(self):
...
+                        'pron_dict_name': data['pron_dict_name']
                     }
-                    models.append(model_info)
+                    models.append(model)
         return models
 
     def new_transcription(self, tname):
         if self.engine is None:
-            raise RuntimeError("Engine needs to be set prior to transcription")
+            raise RuntimeError("Engine need to be set prior to transcription")
```
Comment (nicklambourne) [inline elpis/engines/common/objects/interface.py]: This is a very small thing, but it seems to have gone from grammatically correct to not?

**Comment by benfoley** [comment]
While adding types I also removed a setting value that was called "type". It was being used to do two things: first to set the type of UI element and also to hold info extracted from training files to build select options (eg with tier names etc). I removed this "type" value and added two separate parameters—ui_format and options. Now ui_format is used to determine the input component type, and options is used to hold file values and build the select options.

File: elpis/transformer/__init__.py [hunk 4]
```diff
@@ -69,15 +66,15 @@ class DataTransformer:
...
-                    callback_get_config,
-                    callback_set_config,
-                    settings_change_callback: SettingsChangeCallback
-                    ):
-
-        # Varialbles to be saved between in non-volitile memory exist in the config
+                 name: str,
+                 settings: Dict,
+                 ui: List,
+                 callback_get_config,
```
Comment (nicklambourne) [inline elpis/transformer/__init__.py:72]: Should these have types, too?

File: elpis/transformer/__init__.py [hunk 5]
```diff
@@ -419,133 +415,150 @@ def export(self, f: Callable):
...
-            return 'int'
-        if isinstance(t, list):
-            # TODO: check the types in t
-            return t
-        print("t:", t, ":", type(t), ': t is list =', t is list, ": type(t) == list = ", type(t) == list)
-        raise ValueError(f'type \'{t}\' is not a valid type')
-
-    def import_setting(self, key, type, ui_format=None, default=None, display_name=None, description=None):
+
+    def import_setting(self,
```
Comment (nicklambourne) [inline elpis/transformer/__init__.py:419]: Missing types

File: elpis/transformer/__init__.py [hunk 6]
```diff
@@ -419,133 +415,150 @@ def export(self, f: Callable):
...
             'display_name': display_name,
-            'description': description
+            'description': description,
+            'options': options
         }
         self._import_ui_type_config[key] = 'setting'
         self._import_ui_order_config.append(key)
     
-    def export_setting(self, key, type, ui_format=None, default=None, display_name=None, description=None):
+    def export_setting(self,
```
Comment (nicklambourne) [inline elpis/transformer/__init__.py:449]: Missing types

File: elpis/transformer/__init__.py [hunk 7]
```diff
@@ -419,133 +415,150 @@ def export(self, f: Callable):
...
-                        get_config_callback,
-                        set_config_callback,
-                        settings_change_callback: SettingsChangeCallback
-                    ) -> DataTransformer:
-
+                       collection_path: str,
+                       resampled_path: str,
+                       temporary_directory_path: str,
+                       transcription_json_file_path: str,
+                       get_config_callback,
```
Comment (nicklambourne) [inline elpis/transformer/__init__.py:558]: Missing types

File: elpis/transformer/__init__.py [hunk 8]
```diff
@@ -741,19 +757,20 @@ def build_exporter(self,
...
-                    get_config_callback,
-                    set_config_callback,
-                    settings_change_callback=_default_settings_change_callback
-
-                ) -> DataTransformer:
+                  collection_path: str,
+                  resampled_path: str,
+                  temporary_directory_path: str,
+                  transcription_json_file_path: str,
+                  get_config_callback,
```
Comment (nicklambourne) [inline elpis/transformer/__init__.py:770]: If these are too hard to type, just mark all of these as resolved.

File: elpis/transformer/__init__.py [hunk 9]
```diff
@@ -770,15 +787,16 @@ def make_importer(name: str,
...
-                    transcription_json_file_path: str,
-                    get_config_callback,
-                    set_config_callback,
-                    settings_change_callback=_default_settings_change_callback
-                ) -> DataTransformer:
+                  collection_path: str,
+                  resampled_path: str,
+                  temporary_directory_path: str,
+                  transcription_json_file_path: str,
+                  get_config_callback,
```
Comment (nicklambourne) [inline elpis/transformer/__init__.py:796]: And again

File: elpis/engines/common/objects/interface.py (hunk 3 above)
Comment (benfoley) [reply elpis/engines/common/objects/interface.py]: fixed

File: elpis/transformer/__init__.py (hunk 4 above)
Comment (benfoley) [reply elpis/transformer/__init__.py:72]: Will have to ask Nic

File: elpis/transformer/__init__.py (hunk 5 above)
Comment (benfoley) [reply elpis/transformer/__init__.py:419]: arg, lost in the merge.. will add

File: elpis/transformer/__init__.py (hunk 6 above)
Comment (benfoley) [reply elpis/transformer/__init__.py:449]: ditto

Comment (benfoley) [reply elpis/transformer/__init__.py:449]: added

File: elpis/transformer/__init__.py (hunk 5 above)
Comment (benfoley) [reply elpis/transformer/__init__.py:419]: added

**Pull request merged**